test/unit_tests/test_ahs_device.py

```diff
@@ -556,33 +559,55 @@ def test_result_to_sample_output(self, res, expected_output):
         assert np.allclose(output, expected_output, equal_nan=True)
 
 
-class TestLocalAquilaDevice:
+class TestLocalAhsDevice:
     """Test functionality specific to the local simulator device"""
 
-    def test_validate_operations_multiple_drive_terms(self):
-        """Test that an error is raised if there are multiple drive terms on
-        the Hamiltonian"""
-        pulses = [HardwarePulse(3, 4, 5, [0, 1]), HardwarePulse(4, 6, 7, [1, 2])]
-
-        with pytest.raises(NotImplementedError, match="Multiple pulses in a Hamiltonian are not currently supported"):
+    @pytest.mark.parametrize(
+        "pulses, error",
+        [
+            (
+                [HardwarePulse(3, 4, 5, [0, 1, 2]), HardwarePulse(4, 6, 7, [1, 0, 2])],
+                "ParametrizedEvolution with multiple global drives",
+            ),
+            (
+                [HardwarePulse(3, 4, 5, [3, 4])],
+                "which are not a subset of device wires",
+            ),
+            (
+                [HardwarePulse(3, 4, 5, [0])],
+                "doesn't apply a global driving field to all wires",
```

Comment:
Device

Reply:
The device. At least as I've understood it, the applied drive is a global one, and we can only modify detuning locally, so there's no way to cancel out the applied drive for a single qubit. I guess if the global drive were _only_ detuning, and then we applied local detuning that exactly cancelled it, so if there's only negative detuning (local shift is positive only), we don't need to apply a global drive. 

But behind the scenes, this would still be uploading a DrivingField (global) and a local field that cancels it out some places. Seems like a very niche case where we should just ask people to set it up that way instead of making special logic for converting in the scenario where there's no global drive but the drive that does exist is identical detuning on some subset of the qubits. 

I'm not sure if its an option, either in our code or on the device, to apply only detuning, without any amplitude.